source/common/quic/active_quic_listener.cc

```diff
@@ -294,6 +294,12 @@ ActiveQuicListenerFactory::ActiveQuicListenerFactory(
                                                        validation_visitor,
                                                        server_preferred_address_config_factory),
             validation_visitor, context_);
+
+    // This tells QUICHE to pretend that the client sent this connection option. In the current
+    // QUICHE implementation, the server preferred address will only be sent if the client sends
+    // this non-standard connection option, which is only sent by QUICHE clients. Setting this
+    // causes QUICHE to always send the server preferred address if configured to do so.
+    quic_config_.SetInitialReceivedConnectionOptions(quic::QuicTagVector{quic::kSPAD});
```

Comment:
Is it possible to hide this behind a false RUNTIME_GUARD? We are using this connection option to do client side experiment. If it become default behavior, the clients outside of the experiment group talking to our server will also act on SPA.

Reply:
I think this only affects the behavior of Envoy as a server and should not affect any clients when talking with any other servers.

Reply:
I'm ok adding a runtime guard if you think we should. There's also some risk that `SetInitialReceivedConnectionOptions` will have unexpected side effects.